net/rhp/contracts.go

```diff
@@ -69,26 +70,26 @@ func (c *Contract) PaymentRevision(amount types.Currency) (types.FileContract, e
 	return fcr, nil
 }
 
-// ClearingRevision returns a new file contract revision with the revision
+// ClearingRevision returns a new file contract with the revision
 // number set to uint64 max, the file fields set to their zero value, and the
 // missed proof outputs set to the valid proof outputs.
-func (c *Contract) ClearingRevision() types.FileContract {
-	fcr := c.Revision
-	fcr.RevisionNumber = math.MaxUint64
-	fcr.Filesize = 0
-	fcr.FileMerkleRoot = types.Hash256{}
-	fcr.MissedHostOutput.Value = fcr.ValidHostOutput.Value
-	fcr.MissedRenterOutput.Value = fcr.ValidRenterOutput.Value
-	return fcr
+func ClearingRevision(fc types.FileContract) (rev types.FileContract) {
+	rev = fc
+	rev.RevisionNumber = math.MaxUint64
+	rev.Filesize = 0
+	rev.FileMerkleRoot = types.Hash256{}
+	rev.MissedHostOutput.Value = rev.ValidHostOutput.Value
+	rev.MissedRenterOutput.Value = rev.ValidRenterOutput.Value
+	return
```

Comment:
need to copy the whole output, not just the value. Also, I'd probably just reuse `fc` instead of introducing a new variable